nx_parallel/utils/chunk.py

```diff
@@ -51,3 +65,27 @@ def create_iterables(G, iterator, n_cores, list_of_iterator=None):
         return chunks(list_of_iterator, num_in_chunk)
     else:
         raise ValueError("Invalid iterator type.")
+
+
+def _get_configs(configs):
+    config_dict = asdict(configs)
+    config_dict.update(config_dict["backend_params"])
+    del config_dict["backend_params"]
+    config_dict["n_jobs"] = cpu_count(config_dict["n_jobs"])
+    if "PYTEST_CURRENT_TEST" in os.environ:
+        config_dict["backend"] = "loky"
```

Comment:
Can you help me understand the `PYTEST_CURRENT_TEST` environment variable?

Reply:
here i have just used it to check if we are running pytests, so that i can set the backend to "loky" otherwise with backend=None(default) we get errors from the `joblib.Parallel` call.

ref. https://docs.pytest.org/en/7.1.x/example/simple.html#pytest-current-test-env

Reply:
hmmm....   It feels like we are changing the code when we are testing -- and that makes me uncomfortable. What would happen if someone ran their code with `backend=None`/default? Would that also get errors? I get that this makes it work, but it seems like we should be able to avoid this by setting a default of 'loky' for all ways that call with backend=None. 
:}/

Reply:
> hmmm.... It feels like we are changing the code when we are testing -- and that makes me uncomfortable. 

We were doing this earlier also. We were changing `n_jobs=2` for when we were testing.

> What would happen if someone ran their code with `backend=None`/default? Would that also get errors? 

As mentioned in the previous comment, we do get the following error:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/aditijuneja/Desktop/nx-parallel/nxp-dev/lib/python3.12/site-packages/networkx/utils/decorators.py", line 789, in func
    return argmap._lazy_compile(__wrapper)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<class 'networkx.utils.decorators.argmap'> compilation 20", line 3, in argmap_square_clustering_17
  File "/Users/aditijuneja/Desktop/nx-parallel/nxp-dev/lib/python3.12/site-packages/networkx/utils/backends.py", line 910, in __call__
    return self._convert_and_call(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/aditijuneja/Desktop/nx-parallel/nxp-dev/lib/python3.12/site-packages/networkx/utils/backends.py", line 1322, in _convert_and_call
    result = getattr(backend, self.name)(*converted_args, **converted_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/aditijuneja/Desktop/nx-parallel/nx_parallel/algorithms/cluster.py", line 59, in square_clustering
    with parallel_config(**config):
         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/aditijuneja/Desktop/nx-parallel/nxp-dev/lib/python3.12/site-packages/joblib/parallel.py", line 375, in __init__
    backend = self._check_backend(
              ^^^^^^^^^^^^^^^^^^^^
  File "/Users/aditijuneja/Desktop/nx-parallel/nxp-dev/lib/python3.12/site-packages/joblib/parallel.py", line 439, in _check_backend
    if backend.nesting_level is None:
       ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'nesting_level'
```

> I get that this makes it work, but it seems like we should be able to avoid this by setting a default of 'loky' for all ways that call with backend=None.

I have changed the default backend to `loky` so we would not have to change it from `None` to `"loky"` at the time of testing. Also, now a nx-parallel user will also not get the above error if they don't specify a backend. And that is what happens when a user doesn't specify any backend in the joblib's `parallel_config` context manager because the default backend for the `joblib.Parallel` is `loky`. I'll make sure to document this in the `Config.md` clearly.